source/blog/2015-06-12-ember-1-13-0-released.md

```diff
@@ -0,0 +1,578 @@
+
+For this reason, we've make `mut` objects "actionable". When passed to the
+action helper, they know how to curry themselves into a function where the
+first argument is set as their value. For example, this code sets the
+model's name to "Gretchen" upon click:
+
+```hbs
+{{! app/templates/index.hbs }}
+{{my-component submit=(action (mut model.name))}}
+```
+
+```js
+// app/components/my-component.js
+import Ember from "ember";
+
+export default Ember.Component.extend({
+  click() {
+    this.attrs.submit('Gretchen');
+  }
+});
+```
+
+This nicely isolates the component from knowing whether a mutable value is
+set or some other action is being performed.
+
+Thanks to [@mixonic](https://twitter.com/mixonic) and [@rwjblue](https://twitter.com/rwjblue)
+for their implementation of this feature, as well as to [@wycats](https://twitter.com/wycats)
+and [@KingstonTime](https://twitter.com/KingstonTime) for their thoughtful
+feedback.
+
+#### New Ember.js Helper API
+
+Ember's helper story prior to 1.13 has been inconsistent and neglected. In
+1.13, we're introducing a new API for writing helpers along with a set of
+constraints and features informed by real-world experience.
+
+Ember helpers:
+
+* Represent a single value
+* Do not manage DOM or control flow
+* Can recompute themselves, similar to how a component can rerender
+* Can optionally access services
+* Do not require a dash
+
+Helpers come in two flavors. The first is a function-based API we call a
+shorthand helper. For example, this shorthand helper joins a first and
+last name:
+
+```js
+// app/helpers/full-name.js
+import Ember from "ember";
+
+export default Ember.Helper.helper(function(params, hash) {
+  return params.join(' ');
+});
+```
+
+This helper can be used in a variety of contexts:
+
+```hbs
+{{full-name "Daniel" model.lastName}}
+{{my-component name=(full-name model.firstName "Smith")}}
+{{! The following usage would set the model.name to the new full name
+    when my-component calls the submit action. }}
+{{my-component submit=(action (mut model.name) (full-name model.firstName "Smith"))}}
+```
+
+Helpers receive two arguments: `params` are the ordered params passed to a
+helper, and `hash` contains the key-value options, for example `title="Mr."`.
+
+Some helpers may require access to other parts of Ember (services), and
+some control over their own invalidation and recomputation. In these cases,
+a full class-based helper can be used.
+
+For example, this helper computes a name based on a `name-builder` service. It
+also recomputes whenever the `isAnonymized` state on that service changes:
+
+```js
+// app/helpers/full-name.js
+import Ember from "ember";
+
+export default Ember.Helper.extend({
+  // This service name is only an example
+  nameBuilder: Ember.service.inject(),
+  compute(params, hash) {
+    return this.get('nameBuilder').build(params, hash.title);
+  },
+  rebuildName: Ember.observer('nameBuilder.isAnonymized', function() {
```

Comment:
some funky that we are working to remove observers in components, but they appear here.